test/scripts/unhashKey.ts

```diff
@@ -0,0 +1,179 @@
+                    chosenEndpoint = await select({
+                        message: "Select an endpoint to list pallet prefixes (scroll down for more):",
+                        choices: DEFAULT_ENDPOINTS,
+                    });
+                }
+                let api: ApiPromise;
+                try {
+                    // Wrap connection attempt in a 10 second timeout.
+                    const connectPromise: Promise<ApiPromise> = getApiFor({ ...argv, url: chosenEndpoint });
+                    const timeoutPromise: Promise<ApiPromise> = new Promise((_, reject) =>
+                        setTimeout(() => reject(new Error("Connection timed out after 10 seconds")), 10000)
+                    );
+                    api = await Promise.race([connectPromise, timeoutPromise]);
+                } catch (err) {
+                    console.error(`Failed to connect to ${chosenEndpoint}: ${err}`);
+                    throw err;
+                }
+                try {
+                    console.log("All pallets sorted by storage prefix for chain:", api.runtimeChain.toString());
+                    const pallets = [];
+                    const mt = api.runtimeMetadata;
+                    for (const module of mt.asLatest.pallets) {
+                        if (module.storage.isNone) {
+                            continue;
+                        }
+                        const prefix = xxhashAsHex(module.storage.unwrap().prefix.toString(), 128);
+                        pallets.push({ pallet: module.name.toString(), prefix });
+                    }
+                    // Sort pallets by hex prefix lexicographically.
+                    pallets.sort((a, b) => (a.prefix > b.prefix ? 1 : a.prefix < b.prefix ? -1 : 0));
+                    for (const p of pallets) {
+                        console.log(`${p.prefix} ${p.pallet}`);
+                    }
+                } finally {
+                    await api.disconnect();
+                }
+                return;
+            }
+
+            // Key provided: search mode
+            if (!argv.key.startsWith("0x")) {
+                console.error("Key must start with 0x");
+                process.exit(1);
+            }
+
+            // Determine the list of endpoints to check.
+            // If a URL is provided via the network options, use that only.
+            let endpoints = [];
+            if (argv.url) {
+                endpoints = [argv.url];
+            } else {
+                console.log("No rpc provided. Will try to check all the known endpoints. Use --url to specify one");
+                endpoints = DEFAULT_ENDPOINTS;
+            }
+
+            let found = false;
+            for (const endpoint of endpoints) {
+                console.log(`\nTrying network endpoint: ${endpoint}`);
+                let api: ApiPromise;
+                try {
+                    // Wrap connection attempt in a 10 second timeout.
```

Comment:
We might want to abstract the timeout implementation to the `getApiFor` function with an optional parameter `timeout`.

Reply:
I believe the default timeout is 60 seconds, or more, I will try to change that to 10 as it doesn't make sense to wait that much